samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -439,7 +394,7 @@ object SamzaContainer extends Logging {
       null
 
     val taskFactoryInstance = Option(taskFactory)
```

Comment:
IIUC, taskFactory should never be null at this point. If true, let's remove this variable.